install/firmware.sh.in

```diff
@@ -0,0 +1,104 @@
+#!/bin/sh
+# vim: et sts=4 sw=4
+
+set -eu
+
+firmware="#FIRMWARE#"
+terms_url="https://www.microsoft.com/en-us/legal/terms-of-use"
+
+accept_terms=0
+
+usage() {
+    echo
+    echo "Usage: $(basename "$0") [OPTIONS]"
+    echo
+    echo "Downloads the firmware required by XOW. Saved as $firmware"
+    echo
+    echo "OPTIONS"
+    echo
+    echo "  --accept-terms"
+    echo "        The firmware is covered by the Microsoft Terms of Use."
+    echo "        See the following URL for more details:"
+    echo "        $terms_url"
+    echo
+    echo "  --help"
+    echo "        This help screen"
+    echo
+}
+
+parse_args() {
+    while [ $# -gt 0 ]; do
+        case "$1" in
+            --accept-terms ) shift; accept_terms=1;;
+            -h | --help ) usage; exit 0;;
+            * ) usage; exit 1;;
+        esac
+    done
+}
+
+fail() { echo "$1" >&2; exit 1; }
+
+check_program() {
+    [ -x "$(command -v $1)" ] || fail "Cannot find $1 - required to $2 the firmware ${3:-}."
+}
+
+check_root() {
+    [ "$(id -u)" -eq 0 ] || fail "You have to be root to run this script."
+}
+
+check_programs() {
+    check_program "curl" "download"
+    check_program "cabextract" "extract"
+}
+
+disclaimer() {
+    echo
+    echo "Firmware is distributed under the Microsoft Terms of Use as seen in"
+    echo "$terms_url"
+    echo
+    echo
+    echo "Do you accept the terms of the agreement? [y/N]"
+    echo
+
+    while true; do
+        read -r option
+        case "$option" in
+            Y|YES|y|yes) break;;
+            N|NO|n|no|'') fail "You must agree with the term in order to continue. Exiting";;
+            *) echo "Please answer yes or no";;
+        esac
+    done
+}
+
+get_firmware() {
+    url="http://download.windowsupdate.com/c/msdownload/update/driver/drvs/2017/07/1cd6a87c-623f-4407-a52d-c31be49e925c_e19f60808bdcbfbd3c3df6be3e71ffc52e43261e.cab"
+    sum="48084d9fa53b9bb04358f3bb127b7495dc8f7bb0b3ca1437bd24ef2b6eabdf66"
+    cwd=$(pwd)
+    tmp=$(mktemp -d)
```

Comment:
IMO we should just download the driver package to the current directory instead of creating a temporary folder.

Reply:
agreed, but I think it would still be best to do the extracting in a temporary directory

Reply:
Since things may fail in arbitrary point (hey my internet died during download), the better approach is to operate in tempdir. Otherwise you end with garbage in `pwd`.

Nevertheless, feel free to tweak these as you see fit.